charts/gsp-cluster/templates/02-gsp-system/egress-networkpolicy.yaml

```diff
@@ -18,5 +18,16 @@ spec:
       - 2223
       - 2224
       - 2225
+  - action: Deny
+    protocol: TCP
+    source:
+      namespaceSelector: namespace notin (verify-metadata-controller, verify-proxy-node-build, verify-proxy-node-integration, verify-proxy-node-prod)
```

Comment:
hardcoding these values here feels wrong. Could we instead:

1. check for a `talksToHsm:true` label on the `Namespace`, same way the pod-level check works.
1. extend the gsp-cluster namespace template to allow setting namespace labels from the values.yaml

this would allow configuring which namespaces can talk to hsm by altering the configuration in verify-cluster-config ie:

```
namespaces:
- name: verify-proxy-node-prod
  owner: alphagov
  repository: verify-proxy-node
  path: ci/prod
  requiredApprovalCount: 2
  labels:
    talksToHsm: "true"                             <-- allow adding labels
```

Reply:
I agree! That’s why I wrote the last two paragraphs of this PR 😇

Reply:
I agree! That’s why I wrote the last two paragraphs of this PR 😇